src/threadinterrupt.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <mutex>
+#include <condition_variable>
+#include <atomic>
+
+class CThreadInterrupt
+{
+public:
+    CThreadInterrupt(std::condition_variable& condIn, std::mutex& mutIn) : cond(condIn), mut(mutIn), val(false) {}
+    void reset()
```

Comment:
Sure. Though some of these need to be inlined. I figured someone would yell if I added a .cpp with 2 functions :p